### PR TITLE
[ML] Update the inference schema model definition unit test

### PR DESCRIPTION
**Pull request by valeriy42**
Following changes of @benwtrent in the inference model definition, this PR updates the definition file in the corresponding unit test.

### CODE REVIEW

File: lib/api/CInferenceModelDefinition.cc [hunk 1]
```diff
@@ -278,11 +278,7 @@ std::string CInferenceModelDefinition::jsonString() {
 }
 
 void CInferenceModelDefinition::addToDocument(rapidjson::Value& parentObject,
-                                              TRapidJsonWriter& writer) const { //input
-    rapidjson::Value inputObject = writer.makeObject();
-    m_Input.addToDocument(inputObject, writer);
-    writer.addMember(JSON_INPUT_TAG, inputObject, parentObject);
```
Comment (tveasey) [inline lib/api/CInferenceModelDefinition.cc]: `JSON_INPUT_TAG` is no longer needed I'd guess.

File: lib/api/CInferenceModelDefinition.cc [hunk 2]
```diff
@@ -368,12 +364,8 @@ void CTrainedModel::classificationWeights(const TDoubleVec& classificationWeight
     m_ClassificationWeights = classificationWeights;
 }
 
-void CInferenceModelDefinition::fieldNames(TStringVec&& fieldNames,
-                                           std::size_t dependentVariableColumnIndex) {
-    m_FieldNames = fieldNames;
-    fieldNames.erase(fieldNames.begin() +
-                     static_cast<std::ptrdiff_t>(dependentVariableColumnIndex));
```
Comment (tveasey) [inline lib/api/CInferenceModelDefinition.cc]: Question: does it not matter we include control field names here now, i.e. fields `.`?

Comment (valeriy42) [reply lib/api/CInferenceModelDefinition.cc]: It does not, since we only return the field names actually used by the models.

Comment (tveasey) [reply lib/api/CInferenceModelDefinition.cc]: This is a nice simplification in that case!

File: lib/api/CInferenceModelDefinition.cc (hunk 1 above)
Comment (valeriy42) [reply lib/api/CInferenceModelDefinition.cc]: fixed

**Pull request merged**